server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationIT.java

```diff
@@ -1425,4 +1432,34 @@ public void testIndexWhileRecoveringReplica() throws Exception {
             .get();
         assertNoFailures(response);
     }
+
+    /**
+     * Tests whether segment replication supports realtime get requests and reads and parses source from the translog to serve strong reads.
+     */
+    public void testRealtimeGetRequests() {
+        final String primary = internalCluster().startDataOnlyNode();
+        final String replica = internalCluster().startDataOnlyNode();
+
+        assertAcked(
+            prepareCreate(INDEX_NAME).setSettings(Settings.builder().put("index.refresh_interval", -1).put(indexSettings()))
+                .addAlias(new Alias("alias"))
+        );
+        ensureGreen(INDEX_NAME);
+
+        GetResponse response = client().prepareGet(indexOrAlias(), "1").get();
+        assertThat(response.isExists(), equalTo(false));
+
+        logger.info("--> index doc 1");
+        client().prepareIndex(indexOrAlias()).setId("1").setSource("foo", "bar").get();
+
+        logger.info("--> non realtime get 1");
+        response = client().prepareGet(indexOrAlias(), "1").setRealtime(false).get();
```

Comment:
lets pls specify the replica here and in the next req when intializing the client to ensure the replica is serving the req. `client(replica)` with the 'only_local' preference as well.

Reply:
Thanks for the review @mch2! All valid comments, I'll make the changes shortly